packages/swing-store-lmdb/src/sqlStreamStore.js

```diff
@@ -0,0 +1,148 @@
+// @ts-check
+import sqlite3ambient from 'better-sqlite3';
+import { assert, details as X, q } from '@agoric/assert';
+
+const STREAM_START = { itemCount: 0 };
+
+/**
+ * @param { unknown } streamName
+ * @returns { asserts streamName is string }
+ */
+function insistStreamName(streamName) {
+  assert.typeof(streamName, 'string');
+  assert(streamName.match(/^[-\w]+$/), X`invalid stream name ${q(streamName)}`);
+  return undefined;
+}
+
+/**
+ * @param {unknown} position
+ * @returns { asserts position is StreamPosition }
+ * @typedef {{ itemCount: number }} StreamPosition
+ */
+function insistStreamPosition(position) {
+  assert.typeof(position, 'object');
+  assert(position);
+  assert.typeof(position.itemCount, 'number');
+  assert(position.itemCount >= 0);
+  return undefined;
+}
+
+/**
+ * @param {string} dbDir
+ * @param {{ sqlite3?: typeof import('better-sqlite3') }} [io]
+ */
+export function sqlStreamStore(dbDir, io) {
+  const { sqlite3 = sqlite3ambient } = io || {};
+  const filePath = `${dbDir}/streams.db`; // ISSUE: no path.join?
+  const db = sqlite3(
+    filePath,
+    // { verbose: console.log },
+  );
+  db.exec(`
+    create table if not exists streamItem (
+      streamName text,
+      position integer,
+      item text,
+      primary key (streamName, position)
+    )
```

Comment:
How hard would it be to add a `version` table and insert a `1` into it? Or, how hard is it to (later) sense the lack of a `version` table, so we can treat that as version 0? Every time I've used sqlite I've had the version ID in there from the very beginning.

Reply:
I decided to omit this for now, hoping that "sense the lack of a `version` table" is easy.